### PR TITLE
Introduce tpolecat sbt plugin, resolve compiler warnings

### PR DESCRIPTION
**Pull request by gheine**

### CODE REVIEW

File: app/io/flow/play/actors/ErrorHandler.scala [hunk 1]
```diff
@@ -59,6 +59,7 @@ trait ErrorHandler {
       case t: Throwable =>
         log.error(description.toString, t)
     }
+    ()
```
Comment (mbryzek) [inline app/io/flow/play/actors/ErrorHandler.scala]: seems odd to have to explicitly return `()` everywhere...

File: app/io/flow/play/util/CacheWithFallbackToStaleData.scala [hunk 2]
```diff
@@ -84,7 +85,7 @@ trait CacheWithFallbackToStaleData[K, V] {
     sys.error(s"failureFromEmpty for key[$key]")
   }
 
-  private[this] def failureFromRefresh(key: K, currentEntry: CacheEntry[V], ex: Throwable): CacheEntry[V] = {
```
Comment (jlogeart) [inline app/io/flow/play/util/CacheWithFallbackToStaleData.scala]: I believe a recent change removed the logging of the exception

Comment (jlogeart) [reply app/io/flow/play/util/CacheWithFallbackToStaleData.scala]: Culprit: https://github.com/flowcommerce/lib-play/pull/260/files#diff-1b5bf4e2edfce58b3a37c34afa68a80bR90

File: app/io/flow/play/actors/ErrorHandler.scala (hunk 1 above)
Comment (gheine) [reply app/io/flow/play/actors/ErrorHandler.scala]: The "problem" here is that `f` returns `Any` and `withErrorHandler` returns `Unit` and so the compiler (via the "-Ywarn-value-discard" compiler flag enabled by tpolecat) complains that there's a discarded non-Unit value. By making the expression a statement, that's followed by `()` - a Unit expression - the compiler doesn't detect this as a discarded value any longer. A more correct fix would be to change the signature of `withErrorHandler` to `Any`, or even better introduce a type-parameter, as `def withErrorHandler[T](description: Any)(f :=> T): T`

**Pull request merged**